mono/mini/mini-runtime.c

```diff
@@ -757,7 +757,14 @@ mono_get_lmf (void)
 void
 mono_set_lmf (MonoLMF *lmf)
 {
+	/*
+	 * TLS access resets the last error. This is called from several places
+	 * in the interpreter and debugger where we need to preserve the value (eg.
+	 * to record it for P/Invoke calls).
+	 */
+	W32_DEFINE_LAST_ERROR_RESTORE_POINT;
```

Comment:
Only needed for MinGW.

Reply:
That's what I thought. I checked again and apparently the code is not using `__declspec(thread)` on MSVC here either. It boils down to this:

https://github.com/mono/mono/blob/cad60fce03f1014f0a6b7794d8ca9e28afcc78aa/mono/utils/mono-tls.c#L298-L301

If that is reworked then this can be changed to avoid it on MSVC. I am trying to make a minimal fix and possibly improve upon it later.

Reply:
winconfig.h:

```
/* Have __thread keyword */
#define MONO_KEYWORD_THREAD __declspec (thread)
```

mono-tls.c

```
#ifdef MONO_KEYWORD_THREAD
#define MONO_TLS_GET_VALUE(tls_var,tls_key) (tls_var)
#define MONO_TLS_SET_VALUE(tls_var,tls_key,value) (tls_var = value)
#else
#define MONO_TLS_GET_VALUE(tls_var,tls_key) (mono_native_tls_get_value (tls_key))
#define MONO_TLS_SET_VALUE(tls_var,tls_key,value) (mono_native_tls_set_value (tls_key, value))
#endif
```

Unless msvc is getting the MinGW config.h or something. ?

Reply:
Ah, you are right.